controllers/common/common.go

```diff
@@ -245,6 +245,74 @@ func FullNameForPolicy(plc *policiesv1.Policy) string {
 	return plc.GetNamespace() + "." + plc.GetName()
 }
 
+// GetRepPoliciesInPlacementBinding returns a list of the RepPolicies that are either direct subjects of
+// the given PlacementBinding, or are in PolicySets that are subjects of the PlacementBinding.
+// The list items are guaranteed to be unique (for example if a policy is in multiple sets).
+func GetRepPoliciesInPlacementBinding(
+	ctx context.Context, c client.Client, pb *policiesv1.PlacementBinding,
+) []reconcile.Request {
+	result := make([]reconcile.Request, 0)
+
+	decisions, err := GetDecisions(c, pb) //nolint:contextcheck
+	if err != nil {
+		return result
+	}
+
+	for _, subject := range pb.Subjects {
+		if subject.APIGroup != policiesv1.SchemeGroupVersion.Group {
+			continue
+		}
+
+		switch subject.Kind {
+		case policiesv1.Kind:
+			for _, pd := range decisions {
+				result = append(result, reconcile.Request{NamespacedName: types.NamespacedName{
+					Name:      pb.GetNamespace() + "." + subject.Name,
+					Namespace: pd.ClusterName,
+				}})
+			}
+
+		case policiesv1.PolicySetKind:
+			setNN := types.NamespacedName{
+				Name:      subject.Name,
+				Namespace: pb.GetNamespace(),
+			}
+
+			policySet := policiesv1beta1.PolicySet{}
+			if err := c.Get(ctx, setNN, &policySet); err != nil {
+				continue
+			}
+
+			for _, plc := range policySet.Spec.Policies {
+				for _, pd := range decisions {
+					result = append(result, reconcile.Request{NamespacedName: types.NamespacedName{
+						Name:      pb.GetNamespace() + "." + string(plc),
+						Namespace: pd.ClusterName,
+					}})
+				}
+			}
+		}
+	}
+
+	var unique []reconcile.Request
+
+	table := map[reconcile.Request]bool{}
```

Comment:
Sorry, maybe my misunderstood, this map is for getting unique element

Reply:
Right, so if you set it directly in the map you can skip the intermediary of `result` since you'll be guaranteeing uniqueness as you add the items in.

Reply:
ohoh I see what you mean! Thanks let me try